client/python/tests/test_client_cache.py

```diff
@@ -0,0 +1,213 @@
+from mock import *
+from client_cache import *
+
+import urllib3
+from tx_clock import *
+
+class TestClientCache(object):
+
+    def __init__(self, server, port=80):
+        self.server = server
+        self.port = port
+        self.tests = [
+            self.test_client_cache_init,
+            self.test_client_cache_connection,
+            self.test_client_cache_read,
+            self.test_client_cache_write]
+
+    # Test that we can create a ClientCache instance successfully
+    def test_client_cache_init(self):
+        print "test_client_cache_init",
+
+        server = self.server
+        max_age = 10
+
+        # Default headers
+        cache_default = ClientCache(server)
+        assert(cache_default != None)
+
+        # max_age header
+        cache_max_age = ClientCache(server, 
+            max_age=max_age)
+        assert(cache_max_age != None)
+
+        # no_cache header
+        cache_no_cache = ClientCache(server, no_cache=True)
+        assert(cache_no_cache != None)
+
+        # max_age and no_cache headers
+        cache_both = ClientCache(server, 
+            max_age=max_age, no_cache=True)
+        assert(cache_both != None)
+
+        print "PASSED!"
+
+    # Test that the ClientCache has a valid connection to the server
+    def test_client_cache_connection(self):
+        print "test_client_cache_connection",
+
+        server = self.server
+        max_age = 10
+
+        cache = ClientCache(server, 
+            max_age=max_age)
+        result = cache.pool.request('GET', '/')
+        assert(result != None)
+
+        print "PASSED!"
+
+    # Test that calling read generates the expected request with headers
+    def test_client_cache_read(self):
```

Comment:
I'm a bit confused about what the different test..read functions are each trying to achieve. I'd recommend that test names describe the input and expected behavior. Something like `Class_Inputs_Behavior`.

http://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html


http://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html